config/routes.rb

```diff
@@ -1,8 +1,17 @@
 Rails.application.routes.draw do
   
+  resources :posts
+  resources :categories
+  post '/posts/approve/:id' => 'posts#approve', as: :approve_posts
+  post '/posts/disapprove/:id' => 'posts#disapprove', as: :disapprove_posts
   root to: 'homes#index'
+  get 'users/index'
+  get 'users/show'
   get 'users/edit'
-  patch 'users/update'
+  get 'users/edit_password/:id' => 'users#edit_password', as: :edit_password
+  post 'users/update_pwd/:id' => 'users#update_pwd', as: :update_pwd
+  
+  patch 'users/update/:id' => 'users#update', as: :update
```

Comment:
Please read rails routes and re-define this file, read about `member` and `collection`
`post '/posts/approve/:id' => 'posts#approve', as: :approve_posts` should be changed to:
```
  resources :posts do
    post :approve
  end
```
Apply the same to user

https://guides.rubyonrails.org/routing.html#adding-more-restful-actions